src/document/DocumentCommandHandlers.js

```diff
@@ -1596,6 +1597,12 @@ define(function (require, exports, module) {
         ProjectManager.showInTree(MainViewManager.getCurrentlyViewedFile(MainViewManager.ACTIVE_PANE));
     }
 
+    /** Download selected file or folder structure **/
+    function handleFileDownload() {
+        var entry = ProjectManager.getSelectedItem();
+        ArchiveUtils.archiveWithPath(entry._path,function(){});
```

Comment:
That sucks.  Well, since it takes a `Blob` vs. a BlobURL (i.e., it wants to create the URL for the Blob itself), you'll just have to do that instead (e.g., read the file's contents, get its mime type, and create a new `Blob` yourself).